src/lib/shadow-shim-helper-rs/src/signals.rs

```diff
@@ -0,0 +1,315 @@
+use nix::sys::signal::Signal;
+
+pub const SHD_STANDARD_SIGNAL_MAX_NO: i32 = 31;
+
+/// Lowest and highest valid realtime signal, according to signal(7).  We don't
+/// use libc's SIGRTMIN and SIGRTMAX directly since those may omit some signal
+/// numbers that libc reserves for its internal use. We still need to handle
+/// those signal numbers in Shadow.
+pub const SHD_SIGRT_MIN: i32 = 32;
+pub const SHD_SIGRT_MAX: i32 = 64;
+
+/// Definition is sometimes missing in the userspace headers. We could include
+/// the kernel signal header, but it has definitions that conflict with the
+/// userspace headers.
+pub const SS_AUTODISARM: i32 = 1 << 31;
+
+/// Compatible with the Linux kernel's definition of sigset_t on x86_64.
+///
+/// This is analagous to, but typically smaller than, libc's sigset_t.
+#[repr(C)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct shd_kernel_sigset_t {
+    val: u64,
+}
+
+impl shd_kernel_sigset_t {
+    pub const EMPTY: Self = Self { val: 0 };
+    pub const FULL: Self = Self { val: !0 };
+
+    pub fn has(&self, sig: Signal) -> bool {
+        (*self & shd_kernel_sigset_t::from(sig)).val != 0
+    }
+
+    pub fn lowest(&self) -> Option<Signal> {
+        if self.val == 0 {
+            return None;
+        }
+        for i in 1..=SHD_SIGRT_MAX {
```

Comment:
Since all bits are valid (for example `FULL` has all bits set), should this just loop through all bits (8*sizeof(u64)) rather than some constant, and let `Signal::try_from` fail if it's not a valid signal?